src/dialogs/config-entry-system-options/dialog-config-entry-system-options.ts

```diff
@@ -54,27 +55,31 @@ class DialogConfigEntrySystemOptions extends LitElement {
     await this.updateComplete;
   }
 
+  public closeDialog(): void {
+    this._open = false;
+    this._error = "";
+    this._params = undefined;
+  }
```

Comment:
You should fire the `dialog-closed` event: <https://github.com/home-assistant/frontend/blob/dev/src/dialogs/config-flow/dialog-data-entry-flow.ts#L125>